Create a new community scheduler module for Slinky (Slurm on Kubernetes) (#3862)
GoogleCloudPlatform/cluster-toolkit: examples/README.md

[hpc-slurm-h4d.yaml]: ../community/examples/hpc-slurm-h4d.yaml

### [hpc-slinky.yaml] ![community-badge] ![experimental-badge]

> The SchedMD Slinky Project brings Slurm to Kubernetes, for a best of both worlds experience. Slinky is particularly useful for:

Reviewer: ```suggestion
The SchedMD Slinky Project deploys Slurm on Kubernetes. Slinky is particularly useful for:
```

1. Those with a prefer a Slurm workload management paradigm, but a cloud-native operational experience
2. Those who want the flexibility of running HPC jobs with either Kubernetes-based scheduling or Slurm-based scheduling, all on the same platform

This blueprint creates a simple Slinky installation on top of Google Kubernetes Engine, with the following notable deviations from the Slinky quickstart setup:
1. Two nodesets are implemented, following the pattern of an HPC nodeset and a debug nodeset.
2. A lightweight, GCP-native metrics/monitoring system is adopted, rather than the Slinky-documented cluster-local Kube Prometheus Stack.
3. Node affinities for both system components and compute nodesets are more explicitly defined, to improve stability, control, and HPC hardware utilization.

> While H3 compute-optimized VMs are used for the HPC nodeset, the machine type can easily be switched (including to GPU-accelerated instances).

Reviewer 2: If possible, can we add an example for one of the GPU accelerators in a follow up PR?

Author: Sure thing

> GKE node pools are deliberately autoscaling via `initial_node_count`, rather than `static_node_count`, so that scaling out Slurm nodesets or system components is an easy Kubernetes-/Helm-level update. For example, scaling a debug nodeset from 3 to 5 replicas is a single Nodeset (via Slurm Helm release) config change - no extra work adjusting infrastructure-level specifications. The autoscaling also avoids manual bin packing calculations (especially valuable for nodeset-shared or system component node pools).

Reviewer: This explanation is a bit confusing to me. Does the user have to do anything to autoscale up? Is the slurm nodeset auto-scaling?

Author: The Slurm nodeset is surprisingly not easy to autoscale - requiring additional KEDA work/setup https://github.com/SlinkyProject/slurm-operator/blob/main/docs/autoscaling.md. The example has a non-autoscaling Slurm nodeset, but autoscaling Kubernetes node pool underneath.

Author: Cleaned up this bit of documentation
